controllers/ibmpowervsimage_controller.go

```diff
@@ -97,6 +97,15 @@ func (r *IBMPowerVSImageReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 func (r *IBMPowerVSImageReconciler) reconcile(ctx context.Context, cluster *v1beta1.IBMPowerVSCluster, imageScope *scope.PowerVSImageScope) (ctrl.Result, error) {
 	controllerutil.AddFinalizer(imageScope.IBMPowerVSImage, v1beta1.IBMPowerVSImageFinalizer)
 
+	// Create new labels section for IBMPowerVSImage metadata if nil
+	if imageScope.IBMPowerVSImage.Labels == nil {
+		imageScope.IBMPowerVSImage.Labels = make(map[string]string)
+	}
+
+	if len(imageScope.IBMPowerVSImage.Labels[clusterv1.ClusterLabelName]) == 0 {
```

Comment:
you can use the golang ok statement to check the map entry existence, ref: https://stackoverflow.com/questions/2050391/how-to-check-if-a-map-contains-a-key-in-go